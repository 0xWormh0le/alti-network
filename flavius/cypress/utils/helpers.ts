export const capitalize = (input: string) => {
  if (input.length > 0) {
    const output = input.toLowerCase().split('')
    output[0] = output[0].toUpperCase()
    return output.join('')
  }
  return input
}
