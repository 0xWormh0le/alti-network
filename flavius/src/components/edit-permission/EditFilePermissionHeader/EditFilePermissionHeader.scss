@import 'styles/_shared.scss';

$SEPARATOR_WIDTH: $SMALL_SPACING;

.EditFilePermissionHeader {
  &__title {
    @include flex(row, center, center);
    padding: $SEPARATOR_WIDTH 2 * $SEPARATOR_WIDTH 5px;
    justify-content: space-between;

    &--permissions {
      padding: 0;
    }
  }

  &__edit-permission {
    padding: 5px;

    a {
      width: 120px;
      height: 30px;
      border-radius: 4px;
      border: solid 1px #0177ff;
      padding: 12px 18px;
      font-family: LatoLatin;
      font-size: 10px;
      font-weight: bold;
      font-stretch: normal;
      font-style: normal;
      line-height: 1.8;
      letter-spacing: normal;
      text-align: center;
      color: #0177ff;
      text-decoration: none;

      &:hover {
        color: white;
        background: #0177ff;
      }
    }
  }

  &__right {
    @include flex(row);
  }

  &__subheader {
    @include flex(row, flex-start, center);
    margin: 0 $SEPARATOR_WIDTH $SEPARATOR_WIDTH 2 * $SEPARATOR_WIDTH;

    &--permissions {
      margin: 0;
    }

    &--icon {
      line-height: 1;
      margin-right: 5px;
    }
  }

  &__filename {
    display: inline-block;
    color: $HEADER_COLOR;
    margin: 0;
    max-width: 60vw;

    &:hover {
      color: #008eed;
    }
  }

  &__file-attribute {
    @include set-typography('BODY_SMALL', 'normal');
    color: $TEXT_COLOR_SECONDARY;
    margin-right: 10px;

    &-value {
      margin-left: 4px;
      color: $TEXT_COLOR;
    }
  }

  &__link {
    @include brand-link();
  }

  &__sharing-indicator {
    margin-left: 10px;
  }
}

.SingleFile {
  .EditFilePermissionHeaderLoading {
    margin: $SEPARATOR_WIDTH 2 * $SEPARATOR_WIDTH;
  }
}

.EditFilePermissionHeaderLoading {
  display: flex;
  align-items: flex-start;

  &__content {
    width: 520px;
  }

  &__sharing-indicator {
    margin-left: auto;
  }
}
