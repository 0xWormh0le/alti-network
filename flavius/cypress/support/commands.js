import * as Sentry from '@sentry/browser'
import { RISK_TYPE_IDS } from '../utils/config'
const { MANY_DOWNLOADS_BY_APP_ID } = RISK_TYPE_IDS
import { version } from '../../package.json'

Cypress.on('fail', (error, runnable) => {
  console.info('we failed!!', error, runnable)
  const errorMessage = `Heartbeat Error detected in spec '${runnable.parent.title}': ${error.message}`
  Sentry.init({
    dsn: 'https://4345e62a3abc48049f9965ccc17ed310@sentry.io/1306610',
    environment: Cypress.env('REACT_APP_STAGE'),
    release: `flavius@${version}`,
    integrations: [
      new Sentry.Integrations.Breadcrumbs({
        console: Cypress.env('REACT_APP_STAGE') === 'production',
        message: errorMessage
      })
    ]
  })
  const err = new Error(errorMessage)
  Sentry.withScope((scope) => {
    scope.setExtra('stacktrace', error.stack)
    scope.setExtra('codeframe', error.codeFrame)
    Sentry.captureException(err)
  })
  throw err
})

const checkMinimumValidPerson = (person) => {
  if (person) {
    expect(person.accessCount).to.be.a('number')
    expect(person.primaryEmail?.address).to.be.a('string')
    expect(person.hasOwnProperty('name')).to.be.true
  }
}

const checkMinimumFile = (file) => {
  const {
    createdAt,
    createdBy,
    externalAccessCount,
    externalAccessList,
    fileId,
    fileName,
    internalAccessCount,
    internalAccessList,
    lastModified,
    linkVisibility,
    lastIngested,
    iconUrl
  } = file
  if (createdAt) {
    expect(createdAt).to.be.a('number')
  }
  if (iconUrl) {
    expect(iconUrl).to.be.a('string')
  }
  expect(externalAccessCount).to.be.a('number')
  expect(externalAccessList).to.be.an('Array')
  // todo: uncomment. id and name is currently null
  expect(fileId).to.be.a('string')
  expect(fileName).to.be.a('string')
  expect(internalAccessCount).to.be.a('number')
  expect(internalAccessList).to.be.an('Array')
  if (lastModified) {
    expect(lastModified).to.be.a('number')
  }
  if (lastIngested) {
    expect(lastIngested).to.be.a('number')
  }
  expect(linkVisibility).to.be.a('string')
}

const checkMinimumFiles = (files) => checkMinimumFile(files[randomIndex(files)])

const randomIndex = (array) => Math.floor(Math.random() * array.length)

Cypress.Commands.add('login', () => {
  cy.get('#email').type('bobbie@thoughtlabs.io')
  cy.get('#password').type(Cypress.env('CYPRESS_USER_PASSWORD'))
  cy.get('.Login__button').click()
})

Cypress.Commands.add('checkMinimumValidPerson', checkMinimumValidPerson)

Cypress.Commands.add('checkFullPerson', (person) => {
  checkMinimumValidPerson(person)
  expect(person.externalCount).to.be.a('number')
  expect(person.userKind).to.be.a('number')
  expect(person.internalCount).to.be.a('number')
})

Cypress.Commands.add('checkFiles', (files) => {
  expect(files).to.be.an('Array')
  checkMinimumFiles(files)
})

Cypress.Commands.add('checkRisks', (risks, filteredRiskTypeId) => {
  expect(risks).to.be.an('Array')
  const randomRiskIndex = randomIndex(risks)
  const randomTestRisk = risks[randomRiskIndex]
  expect(randomTestRisk.platformId).to.be.a('string')
  expect(randomTestRisk.datetime).to.be.a('number')
  expect(randomTestRisk.fileCount).to.be.a('number')
  expect(randomTestRisk.hasOwnProperty('fileId')).to.be.true
  expect(randomTestRisk.hasOwnProperty('fileName')).to.be.true
  expect(randomTestRisk.riskDescription).to.be.a('string')
  expect(randomTestRisk.riskId).to.be.a('string')
  expect(randomTestRisk.riskTarget).to.be.an('Array')
  if (filteredRiskTypeId) {
    expect(randomTestRisk.riskTypeId).to.equal(filteredRiskTypeId)
  }
  expect(randomTestRisk.severity).to.be.a('number')

  checkMinimumValidPerson(randomTestRisk.creator)
  checkMinimumValidPerson(randomTestRisk.owner)

  if (filteredRiskTypeId === MANY_DOWNLOADS_BY_APP_ID) {
    expect(randomTestRisk.plugin.id).to.be.a('string')
    // todo: uncomment. name is currently null
    expect(randomTestRisk.plugin.name).to.be.a('string')
  }
})

Cypress.Commands.add('checkPagination', (paginationResponse, overrides, extra) => {
  expect(paginationResponse.orderBy).to.equal(
    overrides && overrides.hasOwnProperty('orderBy') ? overrides.orderBy : 'datetime'
  )
  expect(paginationResponse.pageCount).to.be.a('number')
  expect(paginationResponse.pageNumber).to.be.a('number')
  expect(paginationResponse.pageSize).to.be.a('number')
  expect(paginationResponse.sort).to.equal(overrides && overrides.hasOwnProperty('sort') ? overrides.sort : 'DESC')

  if (extra) {
    const { wrapper } = extra
    const { pageCount } = paginationResponse
    if (pageCount > 1) {
      if (pageCount > 5) {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', 5)
      } else {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', pageCount)
      }
    }
  }
})

Cypress.Commands.add('checkEvents', (events, filesIsObject) => {
  expect(events).to.be.an('Array')
  const randomEventIndex = randomIndex(events)
  const randomTestEvent = events[randomEventIndex]
  expect(randomTestEvent.datetime).to.be.a('number')
  expect(randomTestEvent.eventDescription).to.be.a('string')
  expect(randomTestEvent.eventId).to.be.a('string')
  expect(randomTestEvent.eventName).to.be.a('string')
  expect(randomTestEvent.targetPeople).to.be.an('Array')
  if (!filesIsObject) {
    expect(randomTestEvent.files).to.be.an('Array')
    const randomFileIndex = randomIndex(randomTestEvent.files)
    checkMinimumFile(randomTestEvent.files[randomFileIndex])
  } else {
    checkMinimumFiles(randomTestEvent.files)
  }
})
