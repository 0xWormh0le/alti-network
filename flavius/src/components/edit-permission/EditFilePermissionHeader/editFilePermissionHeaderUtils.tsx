import React from 'react'
import ContentLoader from 'react-content-loader'
import { ellipsify } from 'util/helpers'

export const truncatedFileName = (fileName: string): string => ellipsify(fileName, 60)

export const shouldTruncate = (fileName: string): boolean =>
  fileName !== truncatedFileName(fileName) && truncatedFileName(fileName).indexOf(' ') < 0

interface FileAttributeProps {
  label: string
  value: JSX.Element
}

export const FileAttribute: React.FC<FileAttributeProps> = ({ label, value }) => (
  <span className='EditFilePermissionHeader__file-attribute'>
    <span>{label}</span>
    {value}
  </span>
)

export const EditFilePermissionHeaderLoading = () => (
  <div className='EditFilePermissionHeaderLoading'>
    <ContentLoader
      backgroundColor='#f0f0f0'
      foregroundColor='#f7f7f7'
      height={67}
      width={520}
      className='EditFilePermissionHeaderLoading__content'
      uniqueKey='EditFilePermissionHeaderLoading__content'>
      <rect x={0} y={12} width={520} height={20} rx={4} ry={4} />
      <rect x={0} y={48} width={380} height={15} rx={4} ry={4} />
    </ContentLoader>
    {/* <LinkSharingIndicator className='EditFilePermissionHeaderLoading__sharing-indicator' value='none' loading={true} /> */}
  </div>
)
