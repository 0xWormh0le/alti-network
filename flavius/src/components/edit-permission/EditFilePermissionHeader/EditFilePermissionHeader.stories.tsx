import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import EditFilePermissionHeader, { EditFilePermissionHeaderProps } from './EditFilePermissionHeader'
import { BrowserRouter } from 'react-router-dom'

export default {
  title: 'EditPermission/EditFilePermissionHeader',
  component: EditFilePermissionHeader
} as Meta

const Template: Story<EditFilePermissionHeaderProps> = (args) => (
  <BrowserRouter>
    <EditFilePermissionHeader {...args} />
  </BrowserRouter>
)

export const Default = Template.bind({})
Default.args = {
  file: {
    fileName: 'File',
    createdAt: 123123,
    createdBy: {
      name: {
        givenName: 'Bobbie',
        familyName: '',
        fullName: ''
      }
    },
    lastModified: 456780,
    linkVisibility: 'user'
  } as any
}
