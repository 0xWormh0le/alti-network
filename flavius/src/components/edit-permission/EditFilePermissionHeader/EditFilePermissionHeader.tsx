import React from 'react'
import { Link } from 'react-router-dom'
import AppIndicator from 'components/elements/AppIndicator'
import Person from 'models/Person'
import { modalBasePath, searchWithoutModalParams } from 'util/helpers'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import UI_STRINGS from 'util/ui-strings'
import { formatDate } from 'components/files/FileGridHeader/fileGridHeaderUtils'
import { FileAttribute, shouldTruncate, truncatedFileName } from './editFilePermissionHeaderUtils'
import './EditFilePermissionHeader.scss'

export interface EditFilePermissionHeaderProps {
  file: IFile
}

const EditFilePermissionHeader: React.FC<EditFilePermissionHeaderProps> = ({ file }) => {
  const { fileName, app, fileId, createdBy, createdAt, lastModified, webLink, platformId } = file
  const creator = new Person(createdBy)
  const linkFileInspectorUrl = `${modalBasePath()}/file/${fileId}${searchWithoutModalParams()}`
  const userSpotlightUrl = `${modalBasePath()}/spotlight/${encodeURIComponent(
    creator.primaryEmail?.address ?? '' // altnetId
  )}${searchWithoutModalParams()}`

  return (
    <div className='EditFilePermissionHeader'>
      <div className='EditFilePermissionHeader__title EditFilePermissionHeader__title--permissions'>
        <Link to={linkFileInspectorUrl}>
          <Typography variant={TypographyVariant.H3} weight='normal' className='EditFilePermissionHeader__filename'>
            {shouldTruncate(fileName) ? truncatedFileName(fileName) : fileName}
          </Typography>
        </Link>
      </div>
      <div className='EditFilePermissionHeader__subheader EditFilePermissionHeader__subheader--permissions'>
        <AppIndicator
          value={app}
          fileId={fileId}
          webLink={webLink}
          platformId={platformId}
          className='EditFilePermissionHeader__subheader--icon'
        />
        <FileAttribute
          label={`${UI_STRINGS.EDIT_PERMISSIONS.OWNED_BY}:`}
          value={
            creator.displayName ? (
              <span className='EditFilePermissionHeader__file-attribute-value'>
                <Link className='EditFilePermissionHeader__link' to={userSpotlightUrl}>
                  {creator.displayName}
                </Link>
              </span>
            ) : (
              <span className='EditFilePermissionHeader__file-attribute-value'>
                {UI_STRINGS.EDIT_PERMISSIONS.SHARED_DRIVE}
              </span>
            )
          }
        />
        <FileAttribute
          label={`${UI_STRINGS.FILE.CREATED}:`}
          value={formatDate(createdAt, 'EditFilePermissionHeader__file-attribute-value')}
        />
        <FileAttribute
          label={`${UI_STRINGS.FILE.LAST_MODIFIED}:`}
          value={formatDate(lastModified, 'EditFilePermissionHeader__file-attribute-value')}
        />
      </div>
    </div>
  )
}

export default EditFilePermissionHeader
