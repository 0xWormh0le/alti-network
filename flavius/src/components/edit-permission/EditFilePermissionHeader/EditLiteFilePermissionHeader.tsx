import React from 'react'
import { Link } from 'react-router-dom'
import AppIndicator from 'components/elements/AppIndicator'
import { modalBasePath, searchWithoutModalParams } from 'util/helpers'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import UI_STRINGS from 'util/ui-strings'
import { formatDate } from 'components/files/FileGridHeader/fileGridHeaderUtils'
import { FileAttribute, shouldTruncate, truncatedFileName } from './editFilePermissionHeaderUtils'
import './EditFilePermissionHeader.scss'

export interface EditLiteFilePermissionHeaderProps {
  file: LiteFile
  platformId: string
}

const EditLiteFilePermissionHeader: React.FC<EditLiteFilePermissionHeaderProps> = ({ file, platformId }) => {
  const { fileName, fileId, createdBy, createdAt, lastModified, path } = file
  const linkFileInspectorUrl = `${modalBasePath()}/file/${fileId}${searchWithoutModalParams()}`
  const userSpotlightUrl = `${modalBasePath()}/spotlight/${encodeURIComponent(
    createdBy.primaryEmail?.address ?? ''
  )}${searchWithoutModalParams()}`

  return (
    <div className='EditFilePermissionHeader'>
      <div className='EditFilePermissionHeader__title EditFilePermissionHeader__title--permissions'>
        <Link to={linkFileInspectorUrl}>
          <Typography variant={TypographyVariant.H3} weight='normal' className='EditFilePermissionHeader__filename'>
            {shouldTruncate(fileName) ? truncatedFileName(fileName) : fileName}
          </Typography>
        </Link>
      </div>
      <div className='EditFilePermissionHeader__subheader EditFilePermissionHeader__subheader--permissions'>
        <AppIndicator value={''} fileId={fileId} webLink={path} platformId={platformId} />
        <FileAttribute
          label={`${UI_STRINGS.EDIT_PERMISSIONS.OWNED_BY}:`}
          value={
            createdBy.name.fullName ? (
              <span className='EditFilePermissionHeader__file-attribute-value'>
                <Link className='EditFilePermissionHeader__link' to={userSpotlightUrl}>
                  {createdBy.name.fullName}
                </Link>
              </span>
            ) : (
              <span className='EditFilePermissionHeader__file-attribute-value'>
                {UI_STRINGS.EDIT_PERMISSIONS.SHARED_DRIVE}
              </span>
            )
          }
        />
        <FileAttribute
          label={`${UI_STRINGS.FILE.CREATED}:`}
          value={formatDate(createdAt, 'EditFilePermissionHeader__file-attribute-value')}
        />
        <FileAttribute
          label={`${UI_STRINGS.FILE.LAST_MODIFIED}:`}
          value={formatDate(lastModified, 'EditFilePermissionHeader__file-attribute-value')}
        />
      </div>
    </div>
  )
}

export default EditLiteFilePermissionHeader
