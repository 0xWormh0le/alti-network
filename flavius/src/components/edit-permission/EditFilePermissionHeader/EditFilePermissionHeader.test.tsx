import React from 'react'
import { fileMock } from 'test/mocks'
import EditFilePermissionHeader from './EditFilePermissionHeader'
import { renderWithRouter } from 'test/support/helpers'

describe('EditFilePermissionHeader', () => {
  it('renders correctly', () => {
    const { container } = renderWithRouter(<EditFilePermissionHeader file={fileMock} />)
    expect(container).toMatchSnapshot()
  })
})
