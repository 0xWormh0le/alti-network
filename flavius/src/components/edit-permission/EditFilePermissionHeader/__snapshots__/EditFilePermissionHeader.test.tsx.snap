// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFilePermissionHeader renders correctly 1`] = `
<div>
  <div
    class="EditFilePermissionHeader"
  >
    <div
      class="EditFilePermissionHeader__title EditFilePermissionHeader__title--permissions"
    >
      <a
        href="/dashboard/file/1ghSxcB-ETCezb9odJLeHJufyFrBC_cG3"
      >
        <h3
          class="Typography Typography--h3 Typography--normal EditFilePermissionHeader__filename"
        >
          Quarterly Earnings
        </h3>
      </a>
    </div>
    <div
      class="EditFilePermissionHeader__subheader EditFilePermissionHeader__subheader--permissions"
    >
      <span
        class="AppIndicator EditFilePermissionHeader__subheader--icon"
      >
        <div
          class="platform-container"
        >
          <a
            class="trigger"
            href="https://docs.google.com/document/d/1ghSxcB-ETCezb9odJLeHJufyFrBC_cG3/edit?usp=drivesdk"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div>
              <div>
                <div
                  class="platform-icon-container"
                  data-testid="platform-icon"
                >
                  <img
                    alt="Google Workspace"
                    class="platform-icon"
                    src="google-logo.svg"
                    title="Google Workspace"
                  />
                </div>
              </div>
            </div>
          </a>
        </div>
      </span>
      <span
        class="EditFilePermissionHeader__file-attribute"
      >
        <span>
          Owned by:
        </span>
        <span
          class="EditFilePermissionHeader__file-attribute-value"
        >
          <a
            class="EditFilePermissionHeader__link"
            href="/dashboard/spotlight/test%40email.com"
          >
            Test User
          </a>
        </span>
      </span>
      <span
        class="EditFilePermissionHeader__file-attribute"
      >
        <span>
          Created:
        </span>
        <time
          class="EditFilePermissionHeader__file-attribute-value"
          datetime="1504170378000"
        >
          Aug 31, 2017
        </time>
      </span>
      <span
        class="EditFilePermissionHeader__file-attribute"
      >
        <span>
          Last Modified:
        </span>
        <time
          class="EditFilePermissionHeader__file-attribute-value"
          datetime="1534150370000"
        >
          Aug 13, 2018
        </time>
      </span>
    </div>
  </div>
</div>
`;
